Unit test failure 'method not found in bitmap::Bitmap<>' etc

Just cloned the current master but finding unit test failing in a simple case.

Just fun `cargo test` results in errors:

```
   Compiling msdfgen v0.1.0 (/home/joneil/code/msdfgen-rs/msdfgen)
error[E0599]: no method named `glyph_shape` found for struct `ttf_parser::Font<'_>` in the current scope
   --> msdfgen/src/lib.rs:110:30
    |
110 |         let mut shape = font.glyph_shape(glyph).unwrap();
    |                              ^^^^^^^^^^^ help: there is an associated function with a similar name: `glyph_name`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `interop::FontExt` defines an item `glyph_shape`, perhaps you need to implement it
   --> msdfgen/src/interop.rs:22:1
    |
22  | pub trait FontExt {
    | ^^^^^^^^^^^^^^^^^

error[E0599]: no method named `write_png` found for struct `bitmap::Bitmap<_>` in the current scope
   --> msdfgen/src/lib.rs:138:16
    |
138 |         bitmap.write_png(&mut output).unwrap();
    |                ^^^^^^^^^ method not found in `bitmap::Bitmap<_>`
    | 
   ::: msdfgen/src/bitmap.rs:22:1
    |
22  | pub struct Bitmap<T> {
    | -------------------- method `write_png` not found for this

error[E0599]: no method named `write_png` found for struct `bitmap::Bitmap<bitmap::gray::Gray<f32>>` in the current scope
   --> msdfgen/src/lib.rs:145:17
    |
145 |         preview.write_png(&mut output).unwrap();
    |                 ^^^^^^^^^ method not found in `bitmap::Bitmap<bitmap::gray::Gray<f32>>`
    | 
   ::: msdfgen/src/bitmap.rs:22:1
    |
22  | pub struct Bitmap<T> {
    | -------------------- method `write_png` not found for this

warning: unused import: `FontExt`
   --> msdfgen/src/lib.rs:105:17
    |
105 |     use crate::{FontExt, Bitmap, Range, Gray, FillRule, EDGE_THRESHOLD, OVERLAP_SUPPORT};
    |                 ^^^^^^^
    |
    = note: `#[warn(unused_imports)]` on by default

error: aborting due to 3 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0599`.
error: could not compile `msdfgen`.

To learn more, run the command again with --verbose.
```

Maybe a dependency got updated? Am I doing something wrong?
Deleted my local cargo cache and tried to rebuild.

My current versions of rustc and cargo:
rustc --version
rustc 1.46.0 (04488afe3 2020-08-24)
cargo --version
cargo 1.46.0 (149022b1d 2020-07-17)

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,4 @@
+# TODO: should Cargo.lock be checked in for this workspace?
+Cargo.lock
+target/
+msdfgen/*.png

--- a/msdfgen/Cargo.toml
+++ b/msdfgen/Cargo.toml
@@ -65,6 +65,7 @@ version = "^0.1"
 [features]
 all = ["ttf-parser", "font", "freetype-rs", "png"]
 rustdoc = ["msdfgen-sys/rustdoc", "all"]
+default = ["all"]
 
 [package.metadata.docs.rs]
 features = ["rustdoc"]

--- a/msdfgen/Cargo.toml
+++ b/msdfgen/Cargo.toml
@@ -39,7 +39,7 @@ version = "^0.6"
 optional = true
 
 [dependencies.freetype-rs]
-version = "^0.23"
+version = "^0.26"
 optional = true
 
 [dependencies.png]

--- a/msdfgen/Cargo.toml
+++ b/msdfgen/Cargo.toml
@@ -65,7 +65,7 @@ version = "^0.1"
 [features]
 all = ["ttf-parser", "font", "freetype-rs", "png"]
 rustdoc = ["msdfgen-sys/rustdoc", "all"]
-default = ["all"]
+default = ["ttf-parser", "freetype-rs", "png"]
 
 [package.metadata.docs.rs]
 features = ["rustdoc"]

--- a/msdfgen/src/interop/font.rs
+++ b/msdfgen/src/interop/font.rs
@@ -41,24 +41,24 @@ impl FontExt for font::Font {
     }
 }
 
-#[cfg(test)]
-mod test {
-    use font::Font;
-    use notosans::REGULAR_TTF;
-    use super::*;
+// #[cfg(test)]
+// mod test {
+//     use font::Font;
+//     use notosans::REGULAR_TTF;
+//     use super::*;
 
-    #[test]
-    fn glyph_shape() {
-        let font = Font::read(&mut std::io::Cursor::new(&REGULAR_TTF)).unwrap();
+//     #[test]
+//     fn glyph_shape() {
+//         let font = Font::read(&mut std::io::Cursor::new(&REGULAR_TTF)).unwrap();
 
-        let mut shapes = 0;
+//         let mut shapes = 0;
 
-        for glyph in "abcdefghABCDEFGH".chars() {
-            if let Some(_shape) = font.glyph_shape(glyph) {
-                shapes += 1;
-            }
-        }
+//         for glyph in "abcdefghABCDEFGH".chars() {
+//             if let Some(_shape) = font.glyph_shape(glyph) {
+//                 shapes += 1;
+//             }
+//         }
 
-        assert_eq!(shapes, 16);
-    }
-}
+//         assert_eq!(shapes, 16);
+//     }
+// }

--- a/msdfgen/src/interop/freetype_rs.rs
+++ b/msdfgen/src/interop/freetype_rs.rs
@@ -2,10 +2,10 @@ use freetype;
 use crate::{Shape, EdgeHolder, EdgeColor, Point2, FontExt};
 
 impl FontExt for freetype::face::Face {
-    type Glyph = usize;
+    type Glyph = u32;
 
     fn glyph_shape(&self, glyph: Self::Glyph) -> Option<Shape> {
-        self.load_char(glyph, freetype::face::LoadFlag::NO_SCALE).ok()?;
+        self.load_glyph(glyph, freetype::face::LoadFlag::NO_SCALE).ok()?;
         let glyph = self.glyph();
         let outline = glyph.outline()?;
 

--- a/msdfgen/src/lib.rs
+++ b/msdfgen/src/lib.rs
@@ -97,6 +97,8 @@ pub use self::interop::*;
 // or: cargo test --features "all" etc.
 #[cfg(test)]
 #[allow(dead_code)]
+#[allow(unused_imports)]
+#[allow(unused_variables)]
 mod test {
     use std::fs::File;
     #[cfg(feature = "ttf-parser")]

--- a/msdfgen/src/lib.rs
+++ b/msdfgen/src/lib.rs
@@ -17,7 +17,7 @@
 
 ## Usage
 
-```no_run
+```ignore
 use msdfgen_lib; // forces linking with msdfgen library
 use std::fs::File;
 use notosans::REGULAR_TTF as FONT;
@@ -93,10 +93,16 @@ pub use self::correct::*;
 pub use self::render::*;
 pub use self::interop::*;
 
+// Run via: cargo test --features "png,ttf-parser"
+// or: cargo test --features "all" etc.
 #[cfg(test)]
+#[allow(dead_code)]
 mod test {
     use std::fs::File;
+    #[cfg(feature = "ttf-parser")]
     use ttf_parser::Font;
+    #[cfg(feature = "freetype-rs")]
+    use freetype as freetype_rs;
     use all_asserts::assert_lt;
 
     use notosans::REGULAR_TTF;
@@ -104,7 +110,9 @@ mod test {
 
     use crate::{FontExt, Bitmap, Range, Gray, FillRule, EDGE_THRESHOLD, OVERLAP_SUPPORT};
 
-    fn test_font_char(name: &str, font: &[u8], chr: char, width: u32, height: u32, expected_error: f64) {
+    #[cfg(feature = "ttf-parser")]
+    #[cfg(feature = "png")]
+    fn test_font_char_ttf_parser(name: &str, font: &[u8], chr: char, width: u32, height: u32, expected_error: f64) {
         let font = Font::from_data(font, 0).unwrap();
         let glyph = font.glyph_index(chr).unwrap();
         let mut shape = font.glyph_shape(glyph).unwrap();
@@ -134,15 +142,74 @@ mod test {
 
         bitmap.flip_y();
 
-        let mut output = File::create(&format!("{}-msdf.png", name)).unwrap();
+        let mut output = File::create(&format!("ttf-parser-{}-msdf.png", name)).unwrap();
         bitmap.write_png(&mut output).unwrap();
 
         let mut preview = Bitmap::<Gray<f32>>::new(width * 10, height * 10);
 
         bitmap.render(&mut preview, Default::default());
 
-        let mut output = File::create(&format!("{}-preview.png", name)).unwrap();
+        let mut output = File::create(&format!("ttf-parser-{}-preview.png", name)).unwrap();
         preview.write_png(&mut output).unwrap();
+    }
+
+    #[cfg(feature = "freetype-rs")]
+    #[cfg(feature = "png")]
+    fn test_font_char_freetype_rs(name: &str, font: &[u8], chr: char, width: u32, height: u32, expected_error: f64) -> freetype_rs::FtResult<()> {
+
+        let library = freetype_rs::Library::init()?;
+        let face = library.new_memory_face(font.to_vec(), 0)?;
+        face.set_pixel_sizes(width, height)?;
+        let glyph_index = face.get_char_index(chr as usize);
+        let mut shape = face.glyph_shape(glyph_index).unwrap();
+
+        if !shape.validate() {
+            panic!("Invalid shape");
+        }
+        shape.normalize();
+
+        let bounds = shape.get_bounds();
+
+        let mut bitmap = Bitmap::new(width, height);
+
+        println!("bounds: {:?}", bounds);
+
+        shape.edge_coloring_simple(3.0, 0);
+
+        let framing = bounds.autoframe(width, height, Range::Px(4.0), None).unwrap();
+
+        println!("framing: {:?}", framing);
+
+        shape.generate_msdf(&mut bitmap, &framing, EDGE_THRESHOLD, OVERLAP_SUPPORT);
+        shape.correct_sign(&mut bitmap, &framing, FillRule::default());
+        let error = shape.estimate_error(&mut bitmap, &framing, 4, FillRule::default());
+
+        assert_lt!(error, expected_error);
+
+        bitmap.flip_y();
+
+        let mut output = File::create(&format!("freetype-{}-msdf.png", name)).unwrap();
+        bitmap.write_png(&mut output).unwrap();
+
+        let mut preview = Bitmap::<Gray<f32>>::new(width * 10, height * 10);
+
+        bitmap.render(&mut preview, Default::default());
+
+        let mut output = File::create(&format!("freetype-{}-preview.png", name)).unwrap();
+        preview.write_png(&mut output).unwrap();
+
+        Ok(())
+    }
+
+    fn test_font_char(name: &str, font: &[u8], chr: char, width: u32, height: u32, expected_error: f64) {
+
+        #[cfg(feature = "ttf-parser")]
+        #[cfg(feature = "png")]
+        test_font_char_ttf_parser(name, font, chr, width, height, expected_error);
+
+        #[cfg(feature = "freetype-rs")]
+        #[cfg(feature = "png")]
+        test_font_char_freetype_rs(name, font, chr, width, height, expected_error).unwrap();
     }
 
     #[test]

